File: codeforces/Educational Codeforces Round 86 (Rated for Div. 2)/Road To Zero.py
t = int(input())

while t > 0:
    x, y = [int(q) for q in input().split( )]
    a, b = [int(q) for q in input().split()]
    if 2 * a <= b:
        print ((abs(x) + abs(y)) * a)
    else:
        tot = 0
        if (x >= 0 and y >= 0) or (x <= 0 and y <= 0):
            tot += min(abs(x), abs(y)) * b
            tot += (abs(x) - min(abs(x), abs(y))) * a
            tot += (abs(y) - min(abs(x), abs(y))) * a
        else:
            tot += abs(x) * a
            tot += abs(y) * a
        print(tot)
    t -= 1

File: codeforces/Educational Codeforces Round 86 (Rated for Div. 2)/RoadToZero.java
import java.io.*;
import java.util.*;

public class RoadToZero {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while(t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            long x = Long.parseLong(st.nextToken());
            long y = Long.parseLong(st.nextToken());
            st = new StringTokenizer(br.readLine());
            long a = Long.parseLong(st.nextToken());
            long b = Long.parseLong(st.nextToken());
            long tot = 0;
            if (2 * a <= b) {
                tot = Math.abs(x) * a;
                tot += Math.abs(y) * a;
            } else {
                if ((x >= 0 && y >= 0) || (x <= 0 && y <= 0)) {
                    tot += (Math.min(Math.abs(x), Math.abs(y))) * b;
                    tot += (Math.abs(x) - Math.min(Math.abs(x), Math.abs(y))) * a;
                    tot += (Math.abs(y) - Math.min(Math.abs(x), Math.abs(y))) * a;
                } else {
                    tot = Math.abs(x) * a;
                    tot += Math.abs(y) * a;
                }
            }
            System.out.println(tot);
        }
    }
}